File: drouter-plugin-lib/src/main/java/com/didi/drouter/utils/Logger.java
package com.didi.drouter.utils;

import com.didi.drouter.plugin.RouterSetting;

public class Logger {

    public static void p(Object msg) {
        System.out.println(msg);
    }

    public static void v(Object msg) {
        System.out.println("\033[36m" + msg + "\033[0m");
    }

    public static void d(Object msg) {
        if (RouterSetting.Parse.debug) {
            System.out.println("\033[37m" + msg + "\033[0m");
        }
    }

    public static void w(Object msg) {
        System.out.println("\033[32m" + msg + "\033[0m");
    }

    public static void e(Object msg) {
        System.out.println("\033[31m" + msg + "\033[0m");
    }
}

File: plugins/drouter-plugin/src/main/kotlin/com/didi/drouter/utils/Logger.kt
package com.didi.drouter.utils

import com.didi.drouter.plugin.RouterSetting.Parse

object Logger {
    fun p(msg: Any?) {
        println(msg)
    }

    fun v(msg: Any) {
        println("\u001b[36m$msg\u001b[0m")
    }

    fun d(msg: Any) {
        if (Parse.debug) {
            println("\u001b[37m$msg\u001b[0m")
        }
    }

    fun w(msg: Any) {
        println("\u001b[32m$msg\u001b[0m")
    }

    fun e(msg: Any) {
        println("\u001b[31m$msg\u001b[0m")
    }
}
